### ISSUE TITLE
Closed channels do not get reset in `sctp-proto`

### ISSUE THREAD

**Issue by altonen** [algesten/str0m#469]
I'm using the direct API of `str0m` to open and close outbound data channels. The following code: https://github.com/algesten/str0m/blob/eb0fbe0e01d04e38b8b230a9995951d38447750f/src/channel.rs#L202-L225

allocates the next SCTP ID for an outbound data channel, trying to reuse an old ID when possible. Problem is that while it's trying to reuse an ID, the stream has not been unregistered from SCTP and the following issue happens:

```
<channel for stream 1 opened normally>
2024-03-03 13:50:36.259 str0m::sctp: Stream Open -> Closed
2024-03-03 13:50:36.259 str0m::sctp: Poll Close { id: 1 }
2024-03-03 13:50:36.259 str0m: ChannelClose(ChannelId(0))
...
2024-03-03 13:50:36.262 str0m::channel: Allocate channel id: ChannelId(7)
2024-03-03 13:50:36.262 str0m::channel: Associate stream id ChannelId(7) => 1
2024-03-03 13:50:36.262 str0m::channel: Open stream for: ChannelId(7)
2024-03-03 13:50:36.262 str0m::sctp: New stream 1 (AwaitOpen): open_stream
2024-03-03 13:50:36.262 str0m::sctp: Open stream 1
2024-03-03 13:50:36.262 str0m::sctp: Opening stream 1 failed: ErrStreamAlreadyExists with in-band
````

This happens because while the stream [is marked for closing](https://github.com/algesten/str0m/blob/eb0fbe0e01d04e38b8b230a9995951d38447750f/src/sctp/mod.rs#L305-L309) and later [closed](https://github.com/algesten/str0m/blob/eb0fbe0e01d04e38b8b230a9995951d38447750f/src/sctp/mod.rs#L562-L565), it doesn't get unregistered from `sctp-proto` and when `str0m` tries to reuse the same SCTP ID for a new channel, it triggers [an error](https://github.com/webrtc-rs/sctp-proto/blob/d154eb0c227c8401a9410e4ab5ce332edafceee7/src/association/mod.rs#L647-L654).

I imagine there should be a function `pub fn close_stream(StreamId)` in `sctp-proto`'s `Association` that would properly reset and destroy the stream so that the ID can be reused but wanted to open an issue before working on it in case there is an easier way of dealing with this issue. Any thoughts?

**Comment by algesten** [comment]
Hi @altonen!

I wonder if the problem is that we can reuse the id without opening it again? Either way the idea is that `SctpEvent::Close` would only happen when the id is free to reuse. Is there maybe additional logic we can put in sctp/mod.rs that ensures the id is definitely deallocated before we trigger `SctpEvent::Close`?

**Comment by algesten** [comment]
One fix for this was in #470 – there could be better solutions, but this will do for now.
